sys/include/net/wifi.h

```diff
@@ -0,0 +1,207 @@
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Wi-Fi region Europe
+ */
+#define WIFI_REGION_EUROPE              0
+
+/**
+ * @brief   Wi-Fi region North America
+ */
+#define WIFI_REGION_NORTH_AMERICA       1
+
+/**
+ * @brief   Wi-Fi region Asia
+ */
+#define WIFI_REGION_ASIA                2
+
+#if !defined(CONFIG_WIFI_REGION) || defined(DOXYGEN)
+/**
+ * @brief   Used to select regional Wi-Fi restrictions
+ */
+#define CONFIG_WIFI_REGION              WIFI_REGION_EUROPE
+#endif
+
+/**
+ * @brief   Current Wi-Fi region
+ */
+#define WIFI_REGION                     CONFIG_WIFI_REGION
+
+/**
+ * @brief   2.4 GHz channel 1
+ */
+#define WIFI_2_4_CH_1                   1
+
+/**
+ * @brief   2.4 GHz channel 2
+ */
+#define WIFI_2_4_CH_2                   2
+
+/**
+ * @brief   2.4 GHz channel 3
+ */
+#define WIFI_2_4_CH_3                   3
+
+/**
+ * @brief   2.4 GHz channel 4
+ */
+#define WIFI_2_4_CH_4                   4
+
+/**
+ * @brief   2.4 GHz channel 5
+ */
+#define WIFI_2_4_CH_5                   5
+
+/**
+ * @brief   2.4 GHz channel 6
+ */
+#define WIFI_2_4_CH_6                   6
+
+/**
+ * @brief   2.4 GHz channel 7
+ */
+#define WIFI_2_4_CH_7                   7
+
+/**
+ * @brief   2.4 GHz channel 8
+ */
+#define WIFI_2_4_CH_8                   8
+
+/**
+ * @brief   2.4 GHz channel 9
+ */
+#define WIFI_2_4_CH_9                   9
+
+/**
+ * @brief   2.4 GHz channel 10
+ */
+#define WIFI_2_4_CH_10                  10
+
+/**
+ * @brief   2.4 GHz channel 11
+ */
+#define WIFI_2_4_CH_11                  11
+
+/**
+ * @brief   2.4 GHz channel 12
+ */
+#define WIFI_2_4_CH_12                  12
+
+/**
+ * @brief   2.4 GHz channel 13
+ */
+#define WIFI_2_4_CH_13                  13
+
+/**
+ * @brief   2.4 GHz channel 14
+ */
+#define WIFI_2_4_CH_14                  14
+
+#if WIFI_REGION == WIFI_REGION_EUROPE
+/**
+ * @brief   Lowest Wi-Fi channel
+ */
+#define WIFI_2_4_CH_MIN                 1
+
+/**
+ * @brief   Highest Wi-Fi channel
+ */
+#define WIFI_2_4_CH_MAX                 13
+#elif WIFI_REGION == WIFI_REGION_NORTH_AMERICA
+#define WIFI_2_4_CH_MIN                 1
+#define WIFI_2_4_CH_MAX                 11
+#elif WIFI_REGION == WIFI_REGION_ASIA
+#define WIFI_2_4_CH_MIN                 1
+#define WIFI_2_4_CH_MAX                 14
+#else
+#error "WIFI_REGION undefined"
+#endif
+
+/**
+ * @brief   Number of 2.4 GHz channels
+ */
+#define WIFI_2_4_CH_NUMOF               (2 + (WIFI_2_4_CH_MAX - WIFI_2_4_CH_MIN))
+
+/**
+ * @brief   Type to express a Wi-Fi security mode
+ */
+typedef int8_t wifi_security_mode_t;
+
+/**
+ * @brief   The AP is unprotected
+ */
+#define WIFI_SECURITY_MODE_NONE         0
+
+/**
+ * @brief   The AP is protected with WEP
+ */
+#define WIFI_SECURITY_MODE_WEP_PSK      1
+
+/**
+ * @brief   The AP is protected with WPA
+ */
+#define WIFI_SECURITY_MODE_WPA_PSK      2
+
+/**
+ * @brief   The AP requires user authentication with username and password
+ */
+#define WIFI_SECURITY_MODE_EAP_USER     3
```

Comment:
Please use enumaration type:
```suggestion
/**
 * @brief   Type to express a Wi-Fi security mode
 */
enum {
    WIFI_SECURITY_MODE_NONE,          /**<  AP is unprotected */
    ...
} wifi_security_mode_t;
```

Reply:
Not yet address :wink:

Reply:
[42f76bd](https://github.com/RIOT-OS/RIOT/pull/19387/commits/42f76bde2e1a9182477296ff0e1318486c5ad6ec) ?

Reply:
respective renaming was done in other fixups because they fixup different commits

Reply:
Ah ok, I didn't see this update when I started to review again, sorry.